venues/ACM.org/SIGIR/Badging/python/acm19.py

```diff
@@ -0,0 +1,260 @@
+    'writers': [CONFERENCE_ID],
+    'signatures': [CONFERENCE_ID],
+    'signatories': [CONFERENCE_ID],
+    'members': [],
+})
+
+reviewers_invited = openreview.Group.from_json({
+    'id': REVIEWERS_INVITED_ID,
+    'readers':[CONFERENCE_ID, CHAIRS_ID],
+    'writers': [CONFERENCE_ID],
+    'signatures': [CONFERENCE_ID],
+    'signatories': [CONFERENCE_ID],
+    'members': [],
+})
+
+reviewers_declined = openreview.Group.from_json({
+    'id': REVIEWERS_DECLINED_ID,
+    'readers':[CONFERENCE_ID, CHAIRS_ID],
+    'writers': [CONFERENCE_ID],
+    'signatures': [CONFERENCE_ID],
+    'signatories': [CONFERENCE_ID],
+    'members': [],
+})
+
+# Template for artifacts
+
+artifactSubmission = {
+    'title': {
+        'fieldDisplayLabel': 'Title',
+        'description': 'Title of artifact.',
+        'order': 1,
+        'value-regex': '.{1,250}',
+        'required':True
+    },
+    'authors': {
+        'fieldDisplayLabel': 'Authors',
+        'description': 'Comma separated list of author names. Please provide real names.',
+        'order': 2,
+        'values-regex': "[^;,\\n]+(,[^,\\n]+)*",
+        'required':True
+    },
+    'authorids': {
+        'fieldDisplayLabel': 'Author IDs',
+        'description': '''Comma separated list of author email addresses, lowercased, in the same order as above. For authors with existing OpenReview accounts,
+        please make sure that the provided email address(es) match those listed in the author\'s profile. Please provide real emails.''',
+        'order': 3,
+        'values-regex': "([a-z0-9_\-\.]{2,}@[a-z0-9_\-\.]{2,}\.[a-z]{2,},){0,}([a-z0-9_\-\.]{2,}@[a-z0-9_\-\.]{2,}\.[a-z]{2,})",
+        'required':True
+    },
+    'abstract': {
+        'fieldDisplayLabel': 'Abstract',
+        'description': 'Abstract of paper.',
+        'order': 4,
+        'value-regex': '[\\S\\s]{1,5000}',
+        'required':True
+    },
+    'artifact type': {
+        'fieldDisplayLabel': 'Artifact Type',
+        'description': 'Type of the artifact.',
+        'order': 5,
+        'values-dropdown': [
+            'Code',
+            'Dataset',
+            'User study log book',
+            'Other'
+        ],
+        'required': True
+    },
+    'requested badges': {
+        'fieldDisplayLabel': 'Requested Badges',
+        'description': 'Please select all the badges that you are requesting for.',
+        'order': 6,
+        'values-dropdown': [
+            'Artifacts Available',
+            'Artifacts Evaluated – Functional and Reusable',
+            'Results Replicated',
+            'Results Reproduced'
+        ],
+        'required': True
+    },
+    'pdf': {
+        'fieldDisplayLabel': 'PDF',
+        'description': 'Either upload a PDF file or provide a direct link to your artifact (link must begin with http(s))',
+        'order': 7,
+        'value-regex': 'upload|(http|https):\/\/.+',
+        'required':True
+    }
+}
+
+class ArtifactSubmission(openreview.Invitation):
```

Comment:
you can use openreview. Invitation instead, no need to create a class that is being used once.

Reply:
Resolved.